recipe/meta.yaml

```diff
@@ -1,16 +1,17 @@
 {% set name = "armpl" %}
-{% set version = "21.0" %}
+{% set version = "21.1.1" %}
 
 package:
   name: {{ name }}
   version: {{ version }}
 
 source:
-  - url: https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/21-0-0/RHEL7/arm-performance-libraries_{{ version }}_RHEL-7_gcc-10.2.tar?revision=df0303bf-296f-4f67-a4bf-344a05563bf4
-    folder: armpl
+  url: 'https://developer.arm.com/-/media/Files/downloads/hpc/arm-performance-libraries/{{ version | replace(".", "-") }}/arm-performance-libraries_{{ version }}_manylinux.tar'
+  fn: arm-performance-libraries.tar
```

Comment:
I believe the issue is not that the tar is not getting renamed, but that we are not expecting the tar to automatically unpacked. `conda-build` will extract known compression types for us after downloading and renaming.

An example `work` directory of this build shows:
```
$ ll /mnt/packages/armpl_1642090865668/work
total 145M
-rwxrwxrwx 1 root root 145M Jan  5 13:42 arm-performance-libraries_21.1.1_manylinux.sh
-rw-r--r-- 1 root root 6.8K Jan 13 16:21 build_env_setup.sh
-rwxrw-rw- 1 root root  945 Jan 13 16:21 conda_build.sh
drwxrwxrwx 2 root root   67 Jan  5 13:43 license_terms
-rw-r--r-- 1 root root 1.2K Jan 13 16:21 metadata_conda_debug.yaml
-rwxrwxrwx 1 root root  15K Jan  5 13:42 RELEASE_NOTES.txt
```

Notice that the contents of the tar are already extracted, e.g., `arm-performance-libraries_21.1.1_manylinux.sh`.

If this behavior is not what we want, there have been hacks in the past to name the file to something that `conda-build` will not know, e.g., `arm-performance-libraries`. However, I think it would be better to simply allow the natural behavior, not untar it manually, and simply try:

```
./arm-performance-libraries_*.sh --accept --install-to "${PREFIX}" --force
```

If we want to be more agnostic to the version changes. Unless there is something I am missing.

Reply:
Might also be helpful: [If an extracted archive contains only 1 folder at its top level, its contents will be moved 1 level up, so that the extracted package contents sit in the root of the work folder.](https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html#source-from-tarball-or-zip-archive)

Reply:
That makes sense. What I was trying to avoid was the wonkiness is bash glob expansion. It can pretty easily become really messy. I'll swap it over to your suggestion.